app/assets/javascripts/components/materials_bin/_fetch_data_mixin.js.coffee

```diff
@@ -0,0 +1,39 @@
+# Fetches remote data when component is visible (when @props.visible == true).
+# Data is saved under @state[@dataStateKey].
+# Client class has to define:
+#  - .dataUrl property (string)
+#  - .dataStateKey property (string), name of the state key under which data is saved
+# but it can also define:
+#  - .requestParams property (hash), argument of jQuery.ajax
+#  - .processData() (method), it can process raw AJAX response before state is updated
+window.MBFetchDataMixin =
+  getInitialState: ->
```

Comment:
Hm, I remembered from some tutorial that all the built-in (/ lifecycle) methods can be declared both in mixin and the component and they are merged nicely. I've just tested that for `getInitialState` and it seems to work fine - the state from mixin is merged with state declared in the actual class. Maybe this behaviour has been updated recently.

It fails if you declare a fully custom method or property both in mixin and the class.

Other thing is that this particular `getInitialState` is redundant anyway.